src/codecs.js

```diff
@@ -91,23 +91,39 @@ export const mapLegacyAvcCodecs = function(codecString) {
 export const parseCodecs = function(codecString = '') {
   const codecs = codecString.split(',');
   const result = {};
+  const unknown = [];
 
   codecs.forEach(function(codec) {
     codec = codec.trim();
+    let codecType;
 
     ['video', 'audio'].forEach(function(name) {
       const match = regexs[name].exec(codec.toLowerCase());
 
       if (!match || match.length <= 1) {
         return;
       }
+      codecType = name;
 
       // maintain codec case
       const type = codec.substring(0, match[1].length);
       const details = codec.replace(type, '');
 
       result[name] = {type, details};
     });
+
+    if (!codecType) {
+      unknown.push(codec);
+    }
+
+  });
+
+  unknown.forEach(function(codec) {
+    if (!result.video) {
```

Comment:
It might be worth a quick comment here saying that if there's only one codec and it is unknown, we are defaulting to considering it video only, and if there are two unknown codecs, we consider the first audio and the second video. The first case is less likely than the second, but they both seem to be pretty far in the edge case territory. It may also be worth a TODO to handle unknown codecs differently in the future (instead of guessing).

Reply:
Added some comments and a TODO